packages/forms/src/UIForm/fields/MultiSelectTag/MultiSelectTag.component.js

```diff
@@ -0,0 +1,294 @@
+	}
+
+	/**
+	 * Save tags list ref
+	 * @param tags The tags list ref
+	 */
+	onTagsMount(tags) {
+		this.tags = tags;
+	}
+
+	/**
+	 * Manage suggestion selection
+	 * @param event
+	 * @param focusedItemIndex
+	 * @param newFocusedItemIndex
+	 */
+	onKeyDown(event, { focusedItemIndex, newFocusedItemIndex }) {
+		switch (event.which) {
+		case keycode.codes.enter:
+			if (!this.state.suggestions) {
+				break;
+			}
+			event.preventDefault();
+			// suggestions are displayed and an item has the focus : we select it
+			if (Number.isInteger(focusedItemIndex)) {
+				this.onAddTag(event, { itemIndex: focusedItemIndex });
+			}
+			break;
+		case keycode.codes.down:
+		case keycode.codes.up:
+			event.preventDefault();
+			this.setState({ focusedItemIndex: newFocusedItemIndex });
+			break;
+		default:
+			break;
+		}
+	}
+
+	/**
+	 * Update suggestions on input value change
+	 * @param event The input change event
+	 * @param value The new input value
+	 */
+	onChange(event, { value }) {
+		this.updateSuggestions(value);
+	}
+
+	/**
+	 * Update suggestions on input focus
+	 */
+	onFocus() {
+		this.updateSuggestions();
+	}
+
+	/**
+	 * Add a new tag
+	 * @param event The user event
+	 * @param itemIndex The selected suggestion index
+	 */
+	onAddTag(event, { itemIndex }) {
+		const currentValue = this.state.value;
+		const selectedOption = this.state.suggestions[itemIndex];
+		const isCreation = getNewItemText(currentValue) === selectedOption;
+
+		const newValue = isCreation ? currentValue : selectedOption;
+		const payload = {
+			schema: this.props.schema,
+			value: this.props.value.concat(newValue),
+		};
+		this.props.onChange(event, payload);
+		this.props.onFinish(event, payload);
+		this.updateSuggestions('');
+	}
+
+	/**
+	 * Remove a tag
+	 * @param event The user event
+	 * @param itemIndex The tag index
+	 */
+	onRemoveTag(event, itemIndex) {
+		const value = this.props.value.slice(0);
+		value.splice(itemIndex, 1);
+		const payload = { schema: this.props.schema, value };
+		this.props.onChange(event, payload);
+		this.props.onFinish(event, payload);
+	}
+
+	/**
+	 * Remove all suggestions
+	 */
+	resetSuggestions() {
+		this.setState({
+			suggestions: undefined,
+			focusedItemIndex: undefined,
+		});
+	}
+
+	/**
+	 * Add spaces on input to place it after the tags
+	 */
+	updateSpaces() {
+		if (this.props.value.length) {
+			const lastTag = this.tags.querySelector('div.tc-badge:last-child');
+			const paddingLeft = lastTag.offsetLeft + lastTag.offsetWidth + INPUT_TEXT_INDENT;
+			let paddingTop = lastTag.offsetTop - TAGS_MARGIN_TOP;
+			const toNextLine = (this.input.offsetWidth - paddingLeft) < INPUT_MIN_WIDTH;
+			if (toNextLine) {
+				paddingTop += TAGS_HEIGHT;
+				this.input.style.paddingLeft = '0px';
+			} else {
+				this.input.style.paddingLeft = `${paddingLeft}px`;
+			}
+			this.input.style.marginTop = `${paddingTop}px`;
+		} else {
+			this.input.style.paddingLeft = '0px';
+			this.input.style.paddingTop = '0px';
+		}
```

Comment:
this function ... no way to do better than inline style ?

Reply:
Removed --> Changed implementation